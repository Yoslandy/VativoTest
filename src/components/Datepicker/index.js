import React, { useState } from "react"
import { DropdownDate, DropdownComponent } from "react-dropdown-date"
import PropTypes from "prop-types"

function Datepicker({ label }) {
    // eslint-disable-next-line no-unused-vars
    const [dates, setDate] = useState(null)
    // eslint-disable-next-line no-unused-vars
    const [selectedDate, setSelectedDate] = useState("11-15-2015")
    const formatDate = dat => {
        // formats a JS date to 'mm-dd-yyyy'
        const d = new Date(dat)
        let month = `${d.getMonth() + 1}`
        let day = `${d.getDate()}`
        const year = d.getFullYear()

        if (month.length < 2) month = `0${month}`
        if (day.length < 2) day = `0${day}`
        return [year, month, day].join("-")
    }

    return (
        <>
            <div className="flex flex-col">
                <label className="font-roboto text-[12px] text-neutral-gray-100 uppercase">
                    {label}
                </label>
                <DropdownDate
                    startDate="2012-01-01"
                    endDate="2021-12-31"
                    order={[
                        DropdownComponent.month,
                        DropdownComponent.day,
                        DropdownComponent.year,
                    ]}
                    ids={{
                        month: "select-month",
                        day: "select-day",
                        year: "select-year",
                    }}
                    classes={{
                        dateContainer: "flex",
                        yearContainer: "rounded",
                        monthContainer: "rounded",
                        dayContainer: "rounded",
                        year: "rounded dropdown w-[94px] ",
                        month: "rounded mr-2 dropdown w-[94px]",
                        day: "rounded mr-2 dropdown w-[94px]",
                        yearOptions: "dropdown-option",
                        monthOptions: "dropdown-option",
                        dayOptions: "dropdown-option",
                    }}
                    onDateChange={date => {
                        setDate(date)
                        setSelectedDate(formatDate(date))
                        console.log(date)
                    }}
                    /* onMonthChange={month => {
                        // optional
                        console.log(month)
                        console.log({ date: dates, select: selectedDate })
                    }}
                    onDayChange={day => {
                        // optional
                        console.log(day)
                        console.log({ date: dates, select: selectedDate })
                    }}
                    onYearChange={year => {
                        // optional
                        console.log(year)
                        console.log({ date: dates, select: selectedDate })
                    }} */
                    defaultValues={{
                        year: "YYYY",
                        month: "MM",
                        day: "DD",
                    }}
                />
            </div>
        </>
    )
}

export default Datepicker

Datepicker.propTypes = {
    label: PropTypes.string,
}

Datepicker.defaultProps = {
    label: "",
}
