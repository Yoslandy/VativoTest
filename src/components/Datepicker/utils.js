export const days = Array.from({ length: 31 }, (v, k) => k + 1).map(d => ({
    label: d.toString(),
    value: d,
}))

export const months = [
    "January",
    "February",
    "March",
    "April",
    "May",
    "June",
    "July",
    "August",
    "September",
    "October",
    "November",
    "December",
].map((m, i) => ({
    label: m,
    value: i + 1,
}))

const arrayOfYears = () => {
    const max = new Date().getFullYear()
    const min = max - 9
    const year = []

    // eslint-disable-next-line no-plusplus
    for (let i = max; i >= min; i--) {
        year.push(i)
    }

    const listYear = year.map(y => ({
        label: y.toString(),
        value: y,
    }))

    return listYear
}

export const years = arrayOfYears()
