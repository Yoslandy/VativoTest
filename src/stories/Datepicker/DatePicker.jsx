import React from "react"
import { Formik, Form } from "formik"
import Datepicker from "../../components/Datepicker"

export default function DatePicker() {
    return (
        <Formik initialValues={{}}>
            <Form>
                <Datepicker />
            </Form>
        </Formik>
    )
}
