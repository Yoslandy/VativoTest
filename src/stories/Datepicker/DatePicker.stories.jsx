import React from "react"
import Datepicker from "../../components/Datepicker"

export default {
    title: "Example/Datepicker",
    component: Datepicker,
    argTypes: {
        label: "Contract day",
    },
}

function Template(args) {
    return <Datepicker {...args} />
}

export const Default = Template.bind({})
Default.args = {
    label: "Contract day",
}
