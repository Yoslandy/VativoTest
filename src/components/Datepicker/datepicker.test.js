import React from "react"
import { MemoryRouter } from "react-router-dom"
import renderer from "react-test-renderer"
import { Formik, Form } from "formik"
import Datepicker from "./index"

test("Renders Datepicker correctly", () => {
    const component = renderer.create(
        <MemoryRouter>
            <Formik
                initialValues={{
                    datepicker: "",
                }}
            >
                <Form>
                    <Datepicker label="Contract date" />
                </Form>
            </Formik>
        </MemoryRouter>
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
