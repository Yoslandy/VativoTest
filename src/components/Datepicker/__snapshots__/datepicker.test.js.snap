// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Datepicker correctly 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <label
    className="font-roboto text-xs text-primary-120 uppercase"
  >
    Contract date
  </label>
  <div
    className="Datepicker flex"
  >
    <div
      className="flex flex-col"
    >
      <label
        className="font-roboto text-xs text-primary-120 uppercase"
      >
        
      </label>
      <select
        className="dropdown  Datepicker_day"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
      >
        <option
          className="dropdown-option"
          value={1}
        >
          1
        </option>
        <option
          className="dropdown-option"
          value={2}
        >
          2
        </option>
        <option
          className="dropdown-option"
          value={3}
        >
          3
        </option>
        <option
          className="dropdown-option"
          value={4}
        >
          4
        </option>
        <option
          className="dropdown-option"
          value={5}
        >
          5
        </option>
        <option
          className="dropdown-option"
          value={6}
        >
          6
        </option>
        <option
          className="dropdown-option"
          value={7}
        >
          7
        </option>
        <option
          className="dropdown-option"
          value={8}
        >
          8
        </option>
        <option
          className="dropdown-option"
          value={9}
        >
          9
        </option>
        <option
          className="dropdown-option"
          value={10}
        >
          10
        </option>
        <option
          className="dropdown-option"
          value={11}
        >
          11
        </option>
        <option
          className="dropdown-option"
          value={12}
        >
          12
        </option>
        <option
          className="dropdown-option"
          value={13}
        >
          13
        </option>
        <option
          className="dropdown-option"
          value={14}
        >
          14
        </option>
        <option
          className="dropdown-option"
          value={15}
        >
          15
        </option>
        <option
          className="dropdown-option"
          value={16}
        >
          16
        </option>
        <option
          className="dropdown-option"
          value={17}
        >
          17
        </option>
        <option
          className="dropdown-option"
          value={18}
        >
          18
        </option>
        <option
          className="dropdown-option"
          value={19}
        >
          19
        </option>
        <option
          className="dropdown-option"
          value={20}
        >
          20
        </option>
        <option
          className="dropdown-option"
          value={21}
        >
          21
        </option>
        <option
          className="dropdown-option"
          value={22}
        >
          22
        </option>
        <option
          className="dropdown-option"
          value={23}
        >
          23
        </option>
        <option
          className="dropdown-option"
          value={24}
        >
          24
        </option>
        <option
          className="dropdown-option"
          value={25}
        >
          25
        </option>
        <option
          className="dropdown-option"
          value={26}
        >
          26
        </option>
        <option
          className="dropdown-option"
          value={27}
        >
          27
        </option>
        <option
          className="dropdown-option"
          value={28}
        >
          28
        </option>
        <option
          className="dropdown-option"
          value={29}
        >
          29
        </option>
        <option
          className="dropdown-option"
          value={30}
        >
          30
        </option>
        <option
          className="dropdown-option"
          value={31}
        >
          31
        </option>
      </select>
    </div>
    <div
      className="flex flex-col"
    >
      <label
        className="font-roboto text-xs text-primary-120 uppercase"
      >
        
      </label>
      <select
        className="dropdown  Datepicker_month ml-2"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
      >
        <option
          className="dropdown-option"
          value={1}
        >
          January
        </option>
        <option
          className="dropdown-option"
          value={2}
        >
          February
        </option>
        <option
          className="dropdown-option"
          value={3}
        >
          March
        </option>
        <option
          className="dropdown-option"
          value={4}
        >
          April
        </option>
        <option
          className="dropdown-option"
          value={5}
        >
          May
        </option>
        <option
          className="dropdown-option"
          value={6}
        >
          June
        </option>
        <option
          className="dropdown-option"
          value={7}
        >
          July
        </option>
        <option
          className="dropdown-option"
          value={8}
        >
          August
        </option>
        <option
          className="dropdown-option"
          value={9}
        >
          September
        </option>
        <option
          className="dropdown-option"
          value={10}
        >
          October
        </option>
        <option
          className="dropdown-option"
          value={11}
        >
          November
        </option>
        <option
          className="dropdown-option"
          value={12}
        >
          December
        </option>
      </select>
    </div>
    <div
      className="flex flex-col"
    >
      <label
        className="font-roboto text-xs text-primary-120 uppercase"
      >
        
      </label>
      <select
        className="dropdown  Datepicker_year ml-4"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
      >
        <option
          className="dropdown-option"
          value={2022}
        >
          2022
        </option>
        <option
          className="dropdown-option"
          value={2021}
        >
          2021
        </option>
        <option
          className="dropdown-option"
          value={2020}
        >
          2020
        </option>
        <option
          className="dropdown-option"
          value={2019}
        >
          2019
        </option>
        <option
          className="dropdown-option"
          value={2018}
        >
          2018
        </option>
        <option
          className="dropdown-option"
          value={2017}
        >
          2017
        </option>
        <option
          className="dropdown-option"
          value={2016}
        >
          2016
        </option>
        <option
          className="dropdown-option"
          value={2015}
        >
          2015
        </option>
        <option
          className="dropdown-option"
          value={2014}
        >
          2014
        </option>
        <option
          className="dropdown-option"
          value={2013}
        >
          2013
        </option>
      </select>
    </div>
  </div>
</form>
`;
